### DISCUSSION TITLE
blog/what-is-tproxy/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/rootsongjc.github.io#319] [Announcements]
# blog/what-is-tproxy/

本文介绍了 tproxy 透明代理及其使用方法。

https://jimmysong.io/blog/what-is-tproxy/

**Comment by kevinzu007** [comment]
【使用 tproxy 实现透明代理的步骤如下：】，这个例子写的不太容易看懂，最好举个较为实用场景的例子，这样容易理解写，详细讲明的命令含义

**Reply by rootsongjc** [reply]
你的建议很好，这个例子实际上是来自于 Istio Ambient mesh，可以参考我的在[这一篇博客](https://jimmysong.io/blog/ambient-mesh-l4-traffic-path/)。

**Comment by HuangWendell** [comment]
举例的配置命令是不是有些问题呢？
1. iptables -t mangle -A PREROUTING -p tcp --dport 9080 -j TPROXY --on-port 15001 --on-ip 127.0.0.1 --tproxy-mark 0x1/0x1

2. ip rule add local 0.0.0.0/0 dev lo table 100配置是不是有些问题呢？ip route add local 0.0.0.0/0 dev lo table 100
